stdlib/asm/math/u64.masm

```diff
@@ -673,6 +675,31 @@ export.unchecked_shr
     cswap
 end
 
+# Performs left shift of one unsigned 64-bit integer preserving the overflow and
+# using the pow2 operation.
+# The input value to be shifted is assumed to be represented using 32 bit limbs.
+# The shift value is assumed to be in the range [0, 64).
+# Stack transition looks as follows:
+# [b, a_hi, a_lo, ...] -> [d_hi, d_lo, c_hi, c_lo, ...], where (d,c) = a << b
+# This takes 20 cycles.
+export.overflowing_shl
+    pow2
+    u32split
+    exec.overflowing_mul
+end
+
+# Performs right shift of one unsigned 64-bit integer preserving the overflow and
+# using the pow2 operation.
+# The input value to be shifted is assumed to be represented using 32 bit limbs.
+# The shift value is assumed to be in the range [0, 64).
+# Stack transition looks as follows:
+# [b, a_hi, a_lo, ...] -> [d_hi, d_lo, c_hi, c_lo, ...], where c = a / 2^b, d = a % 2^b
+export.overflowing_shr
+    pow2
+    u32split
+    exec.unchecked_divmod
```

Comment:
This execution as an `adv.u64div` operation, that I couldn't find how many cycles this operation will take. I thought about compare the [`inject_div_result_u64`](https://github.com/maticnetwork/miden/blob/016ba6da2686443ab10f9c91a67610cadbf17a5b/processor/src/operations/decorators/mod.rs#L161) operations with the operations that we've already mapped to cost per cycle, but it seems to be a non-precise way to calculating it. How is the best way to do that?

Reply:
"*Decorators (e.g., DEBUG, ADVICE). As opposed to regular operations, decorators do not advance the clock cycle of the VM, and do not affect the state of the deterministic VM components (e.g., stack, memory)*". 
Is this one of those cases?

Reply:
Yes, `adv.u64div` requires 0 cycles.

Reply:
I also couldn't find `push.adv.1` required cycles. As I understand, it reads from `adv` em push onto stack, thats means it will cost the same as `push.1` (2 cycles)?